cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -1500,11 +1500,17 @@ private predicate linearBoundFromGuard(
   //   1. x <= upperbound(RHS)
   //   2. x >= lowerbound(RHS)
   //
-  // For x != RHS, we create trivial bounds:
-  //
-  //   1. x <= typeUpperBound(RHS.getUnspecifiedType())
-  //   2. x >= typeLowerBound(RHS.getUnspecifiedType())
-  //
+  // For x != RHS, we create bounds based on the size of the type, adjusted according to the RHS bounds:
+  //   1. if lowerbound(RHS) == typeUpperBound(RHS.getUnspecifiedType()) == upperbound(RHS)
+  //      then
+  //        x < lowerbound(RHS)
```

Comment:
I'm certainly buying the soundness of this, but I think it's more clear if we change this line to `x < typeUpperBound(RHS.getUnspecifiedType())` to match the `else` case. That way, it's clear that the only change in the inferred bound is replacing a `<=` with a `<`.

And similarly for the second case below.

Reply:
Agreed. Also, isn't it overly detailed to call `getUnspecifiedType` in pseudocode? What if I just remove that?